packages/amplify-cli/src/extensions/amplify-helpers/resource-status-data.ts

```diff
@@ -0,0 +1,536 @@
+      amplifyMeta: stateManager.getBackendConfig(),
+      currentAmplifyMeta: {},
+    };
+  } else {
+    throw new NotInitializedError();
+  }
+}
+
+//helper: Check if directory has been modified by comparing hash values
+async function isBackendDirModifiedSinceLastPush(resourceName, category, lastPushTimeStamp, hashFunction) {
+  // Pushing the resource for the first time hence no lastPushTimeStamp
+  if (!lastPushTimeStamp) {
+    return false;
+  }
+  const localBackendDir = path.normalize(path.join(pathManager.getBackendDirPath(), category, resourceName));
+  const cloudBackendDir = path.normalize(path.join(pathManager.getCurrentCloudBackendDirPath(), category, resourceName));
+
+  if (!fs.existsSync(localBackendDir)) {
+    return false;
+  }
+
+  const localDirHash = await hashFunction(localBackendDir, resourceName);
+  const cloudDirHash = await hashFunction(cloudBackendDir, resourceName);
+
+  return localDirHash !== cloudDirHash;
+}
+
+//API: calculate hash for resource directory : TBD move to library
+export function getHashForResourceDir(dirPath, files?: string[]) {
+  const options: HashElementOptions = {
+    folders: { exclude: ['.*', 'node_modules', 'test_coverage', 'dist', 'build'] },
+    files: {
+      include: files,
+    },
+  };
+  return hashElement(dirPath, options).then(result => result.hash);
+}
+
+//helper: remove specified resources from list of given resources
+function filterResources(resources, filteredResources) {
+  if (!filteredResources) {
+    return resources;
+  }
+  resources = resources.filter(resource => {
+    let common = false;
+    for (let i = 0; i < filteredResources.length; ++i) {
+      if (filteredResources[i].category === resource.category && filteredResources[i].resourceName === resource.resourceName) {
+        common = true;
+        break;
+      }
+    }
+    return common;
+  });
+  return resources;
+}
+//helper: validate category of the resource
+function resourceBelongsToCategoryList(category, categoryList) {
+  if (typeof category === 'string') {
+    return categoryList.includes(category);
+  } else {
+    return false;
+  }
+}
+//helper: filter resources based on category
+function filterResourceCategory(resourceList, categoryList) {
+  return resourceList ? resourceList.filter(resource => resourceBelongsToCategoryList(resource.category, categoryList)) : [];
+}
+
+//Get the name of the AWS service provisioning the resource
+export function getResourceService(category: string, resourceName: string) {
+  let { amplifyMeta } = getAmplifyMeta();
+  const categoryMeta = amplifyMeta ? amplifyMeta[category] : {};
+  return categoryMeta[resourceName]?.service;
+}
+
+//helper to await results of all function calls
+//TODO: replace with 'await for of'
+async function asyncForEach(array, callback) {
+  for (let index = 0; index < array.length; ++index) {
+    await callback(array[index], index, array);
+  }
+}
+//TODO: replace with some library function
+function capitalize(str) {
```

Comment:
Can we have this function in one file and exported , it seems same functions is in `resource-status-diff.ts`

Reply:
moving it to print , since this is only used for view-functions and labels.